utils/vptr/tests/basic.cc

```diff
@@ -210,3 +210,30 @@ TEST(pointer_mapper, multiple_alloc_free) {
   pMap.clear();
   ASSERT_EQ(pMap.count(), 0u);
 }
+
+TEST(pointer_mapper, default_access) {
+  PointerMapper pMap;
+  {
+    ASSERT_EQ(pMap.count(), 0u);
+    void *myPtr = SYCLmalloc(100 * sizeof(float), pMap);
+    ASSERT_NE(myPtr, nullptr);
+
+    ASSERT_FALSE(PointerMapper::is_nullptr(myPtr));
+    ASSERT_TRUE(PointerMapper::is_nullptr(nullptr));
+    ASSERT_EQ(pMap.count(), 1u);
+
+    cl::sycl::queue q;
+    q.submit([&](cl::sycl::handler &h) {
+      auto accDev = pMap.get_access(myPtr, h);
+      h.single_task<class foo3>([=]() {
+        accDev[0] = 1.0f;
+      });
+    });
+
+    auto hostAcc = pMap.get_access<sycl_acc_rw, sycl_acc_host>(myPtr);
+    ASSERT_EQ(hostAcc[0], 1.0f);
```

Comment:
can you wrap the hostAcc around a explicit scope? ({ .. }) that avoids hostAcc destructor called after the SYCLfree. It may work on the current computecpp implementation but there is no guarantee.

Reply:
Done. 05c45360a1f3058fac648fd8e2c9e7b1963c424b